File: app/src/main/java/com/ericho/coupleshare/mvp/LocationsContract.java
package com.ericho.coupleshare.mvp;

import android.support.annotation.NonNull;

import com.ericho.coupleshare.mvp.presenter.LocationsFilterType;

import java.util.List;

/**
 * Created by steve_000 on 12/6/2017.
 * for project CoupleShare
 * package name com.ericho.coupleshare.mvp
 */

public interface LocationsContract {

    interface View extends BaseView<Presenter> {
        void setLoadingIndicator(boolean active);

        void showLocations(List<Location> locations);


        void showTaskMarkedActive();

        boolean isActive();

        void showNoLocations();

        void showFilteringPopUpMenu();

    }

    interface Presenter extends BasePresenter {
        void result(int requestCode, int resultCode);

        void loadLocations(boolean forceUpdate);

        void openLocationDetails(@NonNull Location requestedLocation);

        LocationsFilterType getFiltering();

        void setFiltering(LocationsFilterType filterType);
    }


}

File: app/src/main/java/com/ericho/coupleshare/mvp/LocationsContract.kt
package com.ericho.coupleshare.mvp

import com.ericho.coupleshare.mvp.presenter.LocationsFilterType

/**
 * Created by steve_000 on 8/7/2017.
 * for project CoupleShare
 * package name com.ericho.coupleshare.mvp
 */
interface LocationsContract {

    interface View : BaseView<Presenter> {
        fun setLoadingIndicator(active: Boolean)

        fun showLocations(locations: List<Location>)



        var isActive: Boolean

        fun showNoLocations()

        fun showErrorMessage(error:String)

        fun showFilteringPopUpMenu()
        fun showConfirmDeleteDialog(location: Location)

    }

    interface Presenter : BasePresenter {
        fun result(requestCode: Int, resultCode: Int)

        fun loadLocations(forceUpdate: Boolean)
        fun openLocationDetails(requestedLocation: Location)

        fun deleteOneLocation(location: Location)

        fun confirmDeleteOneLocation(location: Location)


        var filtering: LocationsFilterType
    }
}